VotingApplication/VotingApplication.Web.Api/Scripts/ResultChart.js

```diff
@@ -0,0 +1,92 @@
+﻿define('ResultChart', ['jquery', 'knockout', 'insight'], function ($, ko, insight) {
+    var chartCounter = 0;
+    function ResultChart($element) {
+        chartCounter++;
+
+        var chart;
+
+        this.drawChart = function (data, options) {
+            if (!data.length) return;
+
+            var dataUnchanged = false;
+            var barCount = 0;
+            for (var n = 0; n < data.length; n++) {
+                barCount += data[n].Data.length;
+                dataUnchanged = dataUnchanged &&
```

Comment:
If we know early on that the data has changed, we should break out of the for loop immediately.